os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/i2s_api.c

```diff
@@ -621,6 +624,13 @@ void i2s_disable(i2s_t *obj, bool is_suspend)
 		AUDIO_SP_TXStart(obj->i2s_idx, DISABLE);
 		if (is_suspend) {
 			AUDIO_SP_Deinit(obj->i2s_idx, obj->direction);
+
+			/* deinit the peripheral clock when suspending to avoid invalid state */
+			if (obj->i2s_idx == I2S2) {
+				RCC_PeriphClockCmd(APBPeriph_SPORT2, APBPeriph_SPORT2_CLOCK, DISABLE);
+			} else {
```

Comment:
Although current board pinmux doesn't support internal codec (I2S0/1), it is still better to specify the i2s idx.
`else if (obj->i2s_idx == I2S3)`